src/PostTypeRegister.php

```diff
@@ -68,9 +68,21 @@ public function get_post_types() {
 
 	public function get_post_type_settings( WP_Post $post ) {
 		$settings = empty( $post->post_content ) || isset( $_GET['mbcpt-force'] ) ? $this->migrate_data( $post ) : json_decode( $post->post_content, true );
-
+		switch ( $settings['icon_type'] ) {
+			case 'dashicons' :
+				$settings['menu_icon'] = $settings['icon'];
+				break;
+			case 'svg':
+				$settings['menu_icon'] = $settings['icon_svg'];
+				break;
+			case 'custom':
+				$settings['menu_icon'] = $settings['icon_custom'];
+				break;
+			default:
+				$settings['menu_icon'] = 'dashicons-admin-generic';
+				break;
+		}
```

Comment:
Tách phần này ra thành 1 hàm parse_icon nhé. Và xem cách làm ở các hàm bên dưới như `parse_support`. Sau khi parse xong thì cần `unset` các param ko cần thiết đi, VD `icon`, `icon_custom`, `icon_svg`.

Reply:
Tách tức là chuyển toàn bộ phần code logic sang hàm mới em. Chứ tách như này thì càng rối.